react/features/calendar-sync/actions.js

```diff
@@ -61,3 +66,112 @@ export function setCalendarEvents(events: Array<Object>) {
         events
     };
 }
+
+/**
+ * Sets the calendar type to be used by web.
+ *
+ * @param {string|undefined} calendarType - The calendar type.
+ * @returns {{
+ *     type: SET_CALENDAR_TYPE,
+ *     calendarType: ?string
+ * }}
+ */
+export function setCalendarType(calendarType: ?string) {
+    return {
+        type: SET_CALENDAR_TYPE,
+        calendarType
+    };
+}
+
+/**
+ * Sends an action to update the current calendar api state in redux.
+ *
+ * @param {number} newState - The new state.
+ * @returns {{
+ *     type: SET_CALENDAR_API_STATE,
+ *     apiState: number
+ * }}
+ */
+export function setCalendarAPIState(newState: ?number) {
+    return {
+        type: SET_CALENDAR_API_STATE,
+        apiState: newState
+    };
+}
+
+/**
+ * Prompts the participant to sign in.
+ *
+ * @returns {function(Dispatch<*>, Function): Promise<string | never>}
+ */
+export function signIn() {
+    return (dispatch: Dispatch<*>, getState: Function): Promise<*> => {
+        const { calendarType } = getState()['features/calendar-sync'];
+        const api = _getCalendarIntegration(
+            calendarType, {
+                dispatch,
+                getState
+            });
+
+        if (!api) {
+            return Promise.reject('No calendar type selected!');
```

Comment:
Would it make sense to do this check even before trying to get the api?

Reply:
Nevermind, I get what you're doing. In case some unexpected type is set you want to error here.